SonarSource/sonarqube-scan-action
# SQCPPGHA-9 Extend action to support C, C++, and Objective-C projects

[SQCPPGHA-9](https://sonarsource.atlassian.net/browse/SQCPPGHA-9)

Related PR on `sonarqube-github-c-cpp` side: https://github.com/SonarSource/sonarqube-github-c-cpp/pull/21

Content of this PR:
- sub-action `install-build-wrapper` to download the Build Wrapper from SonarQube (Server or Cloud) and install it locally
- sub-action `deprecated-c-cpp`, where `sonarqube-github-c-cpp` (and indirectly `sonarcloud-github-c-cpp`) will forward to
- update of the README.md to include examples for C, C++, and Objective-C projects
- introduction of a `sonar-scanner-version` file at the root of the repo, with URLs and SHAs of the Sonar Scanner CLI
- `version_update.yml` workflow, to periodically update `sonar-scanner-version` with the latest URLs and SHAs
- QA tests for sub-actions and scripts
- `actions/cache@v4.0.2` changed into `actions/cache@v4` due to [this deprecation notice](https://github.com/actions/cache/discussions/1510)

[SQCPPGHA-9]: https://sonarsource.atlassian.net/browse/SQCPPGHA-9?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Successful runs of SQS projects (only the ones on Linux since tests use a dockerized SQS, not available on Windows and macOS):
- `linux-cmake-compdb-gh-actions-sq`: https://github.com/antonioaversa/linux-cmake-compdb-gh-actions-sq/actions/runs/12282046431/job/34310241563
- `linux-autotools-gh-actions-sq/actions/runs`: https://github.com/antonioaversa/linux-autotools-gh-actions-sq/actions/runs/12282089038/job/34310281085 (uses `v4` for Sonar Scanner CLI installation and execution)
- `linux-cmake-gh-actions-sq`: https://github.com/antonioaversa/linux-cmake-gh-actions-sq/actions/runs/12282130200/job/34310282674

Successful runs of SQC projects:
- `windows-msbuild-opencppcoverage-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-opencppcoverage-gh-actions-sc/actions/runs/12294180509/job/34309535901
- `linux-cmake-compdb-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-compdb-gh-actions-sc/actions/runs/12294696872/job/34310013914
- `windows-msbuild-vscoverage-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-vscoverage-gh-actions-sc/actions/runs/12294748584/job/34310168837
- `macos-xcode-coverage-gh-actions-sc`: https://github.com/antonioaversa/macos-xcode-coverage-gh-actions-sc/actions/runs/12294845114/job/34310467208
- `linux-cmake-gcovr-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-gcovr-gh-actions-sc/actions/runs/12294937667/job/34310758049
- `linux-cmake-llvm-cov-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-llvm-cov-gh-actions-sc/actions/runs/12294991412/job/34310939193
- `macos-xcode-gh-actions-sc`: https://github.com/antonioaversa/macos-xcode-gh-actions-sc/actions/runs/12295069355/job/34311179045
- `macos-cmake-gh-actions-sc`: https://github.com/antonioaversa/macos-cmake-gh-actions-sc/actions/runs/12295100783/job/34311277827
- `windows-msbuild-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-gh-actions-sc/actions/runs/12295179050/job/34311522465
- `windows-cmake-gh-actions-sc`: https://github.com/antonioaversa/windows-cmake-gh-actions-sc/actions/runs/12295276952/job/34311831313
